tests/e2e/helpers/misc/misc.go

```diff
@@ -143,3 +145,50 @@ func TrimStringFromChar(s, c string) string {
 	}
 	return s
 }
+
+func AddNode(name string, index int, file string) error {
+	// Read live XML configuration
+	fileContent, err := exec.Command("sudo", "virsh", "net-dumpxml", "default").Output()
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal fileContent
+	netcfg := &libvirtxml.Network{}
+	if err := netcfg.Unmarshal(string(fileContent)); err != nil {
+		return err
+	}
+
+	// Add new host
+	// NOTE: we only use one network (IPs[0])
+	host := libvirtxml.NetworkDHCPHost{
+		Name: name,
+		MAC:  "52:54:00:00:00:" + fmt.Sprintf("%02x", index),
+		IP:   "192.168.122." + strconv.Itoa(index+1),
+	}
+	netcfg.IPs[0].DHCP.Hosts = append(netcfg.IPs[0].DHCP.Hosts, host)
+
+	// Marshal new content
+	newFileContent, err := netcfg.Marshal()
+	if err != nil {
+		return err
+	}
+
+	// Re-write XML file
+	if err := os.WriteFile(file, []byte(newFileContent), 0644); err != nil {
+		return err
+	}
+
+	// Update live network configuration
+	xmlValue, err := host.Marshal()
+	if err != nil {
+		return err
+	}
+	if err := exec.Command("sudo", "virsh", "net-update",
```

Comment:
Note: this could be replaced to fully use the `libvirt` lib but it will require more changes and more time-consuming tests to do this.